controllers/apirule_controller.go

```diff
@@ -232,37 +232,40 @@ func (r *APIRuleReconciler) getReconciliation(config processing.ReconciliationCo
 // SetupWithManager sets up the controller with the Manager.
 func (r *APIRuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&gatewayv1beta1.APIRule{}).
-		WithEventFilter(predicate.GenerationChangedPredicate{}).
-		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}).
-		WithEventFilter(&configMapPredicate{Log: r.Log}).
+		// We need to filter for generation changes, because we had an issue that on Azure clusters the APIRules were constantly reconciled.
+		For(&gatewayv1beta1.APIRule{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(&isApiGatewayConfigMapPredicate{Log: r.Log})).
 		Complete(r)
 }
 
 // Updates api status. If there was an error during update, returns the error so that entire reconcile loop is retried. If there is no error, returns a "reconcile success" value.
 func (r *APIRuleReconciler) updateStatusOrRetry(ctx context.Context, api *gatewayv1beta1.APIRule, status processing.ReconciliationStatus) (ctrl.Result, error) {
-	_, updateStatusErr := r.updateStatus(ctx, api, status)
+	_, updateStatusErr := r.updateStatus(ctx, api, status, &r.Log)
```

Comment:
The logger is a field in `APIRuleReconciler`. Why do we pass it in function call?